File: src/main/scala/click/seichi/eategg/IsUuidIgnored.scala
package click.seichi.eategg

import java.util.UUID
import scala.collection.mutable

object IsUuidIgnored {
  private val flags: mutable.Map[UUID, Boolean] = mutable.Map.empty

  def get(playerUuid: UUID): Boolean = flags.getOrElse(playerUuid, false)

  def toggle(playerUuid: UUID): Unit = flags.put(playerUuid, !get(playerUuid))

  def reset(playerUuid: UUID): Unit = flags.put(playerUuid, false)
}

File: src/main/java/click/seichi/eategg/IsUuidIgnored.java
package click.seichi.eategg;

import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

public final class IsUuidIgnored {
  private static final Map<UUID, Boolean> FLAGS = new HashMap<>();

  private IsUuidIgnored() {}

  public static boolean get(UUID playerUuid) {
    return FLAGS.getOrDefault(playerUuid, false);
  }

  public static void toggle(UUID playerUuid) {
    FLAGS.put(playerUuid, !get(playerUuid));
  }

  public static void reset(UUID playerUuid) {
    FLAGS.put(playerUuid, false);
  }
}
